packages/nouns-contracts/contracts/governance/NounsDAOProxyV3.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+/// @title The Nouns DAO proxy contract for V3
+
+/*********************************
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░██░░░████░░██░░░████░░░ *
+ * ░░██████░░░████████░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░██░░██░░░████░░██░░░████░░░ *
+ * ░░░░░░█████████░░█████████░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ * ░░░░░░░░░░░░░░░░░░░░░░░░░░░░░ *
+ *********************************/
+
+// NounsDAOProxyV3.sol is a modified version of NounsDAOProxy.sol, tailored for the DAO's V3.
+// Its main purpose is to support people wishing to deploy V3 directly, without having to deploy V1 first and then upgrade.
+
+// LICENSE
+// NounsDAOProxy.sol is a modified version of Compound Lab's GovernorBravoDelegator.sol:
+// https://github.com/compound-finance/compound-protocol/blob/b9b14038612d846b83f8a009a82c38974ff2dcfe/contracts/Governance/GovernorBravoDelegator.sol
+//
+// GovernorBravoDelegator.sol source code Copyright 2020 Compound Labs, Inc. licensed under the BSD-3-Clause license.
+// With modifications by Nounders DAO.
+//
+// Additional conditions of BSD-3-Clause can be found here: https://opensource.org/licenses/BSD-3-Clause
+//
+//
+// NounsDAOProxy.sol uses parts of Open Zeppelin's Proxy.sol:
+// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/5c8746f56b4bed8cc9e0e044f5f69ab2f9428ce1/contracts/proxy/Proxy.sol
+//
+// Proxy.sol source code licensed under MIT License.
+//
+// MODIFICATIONS
+// The fallback() and receive() functions of Proxy.sol have been used to allow Solidity > 0.6.0 compatibility
+
+pragma solidity ^0.8.6;
+
+import './NounsDAOInterfaces.sol';
+
+contract NounsDAOProxyV3 is NounsDAOProxyStorage, NounsDAOEvents {
+    struct ProxyParams {
```

Comment:
@eladmallel what's the purpose of the struct?

Reply:
@davidbrai to avoid stack too deep in the proxy constructor :)